webapp/Connector/src/app/store/Study.js

```diff
@@ -128,6 +152,25 @@ Ext.define('Connector.app.store.Study', {
     onLoadDocuments : function(documentData) {
         this.documentData = documentData.rows;
         this._onLoadComplete();
+
+        //For using valid downloadable link as an "availability" indicator for NI data
+        this.niLinkValidationCount = 0;
+        this.niDocumentData = this.documentData.filter(function (doc) { return doc.document_type === "Non-Integrated Assay" });
+
+        if (this.niDocumentData.length > 0) {
+            for (var itr = 0; itr < this.niDocumentData.length; itr++) {
+
+                var docIsValidAction = function (doc, status) {
```

Comment:
Seems this function definition can be moved out of the loop and reused.

Reply:
yup, now done